File: 14-longest-common-prefix/14-longest-common-prefix.go
func longestCommonPrefix(strs []string) string {
    lcp := len(strs[0])
    for i := 1; i < len(strs); i++ {
        count := 0
        for j := 0; j < len(strs[i]) && j < len(strs[0]); j++ {
            if strs[0][j] == strs[i][j] && count < lcp {
                count++
            } else {
                break
            }
        }
        if lcp > count {
            lcp = count
        }
    }
    return strs[0][:lcp]
}

File: 14-longest-common-prefix/14-longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        
        index = 0
        count = len(strs)
        pre = ""
        
        while True:
            dic = {}
            for string in strs:
                if len(string) == 0:
                    return ""
                if index < len(string):
                    char = string[index]
                else:
                    return pre
                if char in dic.keys():
                    dic[char] += 1
                else:
                    dic[char] = 1
            if dic[list(dic.keys())[0]] == count:
                pre += list(dic.keys())[0]
            else:
                return pre
            index += 1
                    
                